src/useForm.tsx

```diff
@@ -23,6 +24,22 @@ export function useForm<T extends Record<string, any>>(
   const fields = useMemo(() => createFields(state, setState), [state])
   const validate = useCallback(() => runValidation(state, setState), [state])
 
+  const isEmpty = useMemo(() => {
+    let isEmpty = true
+    forEach<FieldsState<T>>(state, (_, { value }: any) => {
```

Comment:
Is there a better primitive than forEach that would allow for early termination as soon as isEmpty has been set to false?

Reply:
Ah yea should have left a comment about this. I had `forEach` laying around in here because elsewhere we have to traverse every property on an object (e.g. to convert the caller's per-field definitions to our per-field state). 

So this is a lazy re-use, that while theoretically is inefficient, probably won't really be noticeable until we have really big forms (e.g 100 fields).  Still, it'd be nice to short-circuit.

Reply:
It'd be pretty easy to write a `findFirst` or similarly named function that recursively traverses an object's properties but bails as soon as a passed predicate evaluates to `true` -- will add that in a subsequent PR.

Reply:
Sounds good.  

Another sort-of related question I had is if there are default values, isEmpty would always return false right?  Just wondering if that's the desired semantic?

Reply:
Yea -- good question. I consider a form with pre-filled / default values to be non-empty -- e.g. if it's a text box it might start with a value of "foo". But if the user deletes that, then the form becomes "empty". 

For additional context, I added `isEmpty` because it gives us an easy way to determine if we should run validation against an entity in a list or filter it out (b/c it's "empty").